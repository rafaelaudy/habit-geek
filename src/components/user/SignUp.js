import React, { useState } from "react";
import { navigate } from "@reach/router";
import { useTranslation } from "react-i18next";
import * as userService from "@rafael.audy/habit-geek-utils/services/userService";
import UserForm from "./UserForm";

const SignUp = ({ jwt, saveSession }) => {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [validityChecked, setValidityChecked] = useState(false);
  const { t } = useTranslation();

  if (jwt) {
    navigate("/habit-geek/profile");
  }

  const signUp = event => {
    event.preventDefault();
    setValidityChecked(true);

    if (event.target.checkValidity()) {
      userService
        .signUp(email, password)
        .then(saveSession)
        .catch(() => {});
    }
  };

  return (
    <div className="mobile-size-container">
      <form
        className={`${validityChecked ? "was-validated" : ""}`}
        onSubmit={signUp}
        noValidate
      >
        <h2>{t("sign-up-title")}</h2>

        <UserForm
          email={email}
          password={password}
          setEmail={setEmail}
          setPassword={setPassword}
        ></UserForm>

        <hr className="mb-4" />
        <button type="submit" className="btn btn-primary btn-lg btn-block">
          {t("sign-up-sign-up")}
        </button>
      </form>
      <div>
        {t("sign-up-go-to-login-label")}
        <button
          onClick={() => navigate("/habit-geek/login")}
          className="px-0 btn btn-link"
        >
          {t("sign-up-go-to-login")}
        </button>
      </div>
    </div>
  );
};

export default SignUp;
