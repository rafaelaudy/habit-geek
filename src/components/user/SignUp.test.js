import React from "react";
import { shallow } from "enzyme";
import SignUp from "./SignUp";

//TODO
describe("SignUp component", () => {
  it("Renders static elements", () => {
    const signUp = shallow(<SignUp />);
    expect(signUp).toMatchSnapshot();
  });
});
