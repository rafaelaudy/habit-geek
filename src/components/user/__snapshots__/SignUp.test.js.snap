// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp component Renders static elements 1`] = `
<div
  className="mobile-size-container"
>
  <form
    className=""
    noValidate={true}
    onSubmit={[Function]}
  >
    <h2>
      sign-up-title
    </h2>
    <UserForm
      email=""
      password=""
      setEmail={[Function]}
      setPassword={[Function]}
    />
    <hr
      className="mb-4"
    />
    <button
      className="btn btn-primary btn-lg btn-block"
      type="submit"
    >
      sign-up-sign-up
    </button>
  </form>
  <div>
    sign-up-go-to-login-label
    <button
      className="px-0 btn btn-link"
      onClick={[Function]}
    >
      sign-up-go-to-login
    </button>
  </div>
</div>
`;
